dvc/repo/get.py

```diff
@@ -29,32 +30,41 @@ def get(url, path, out=None, rev=None):
     # reflink and/or hardlink. Not using tempfile.TemporaryDirectory
     # because it will create a symlink to tmpfs, which defeats the purpose
     # and won't work with reflink/hardlink.
-    dpath = os.path.dirname(os.path.abspath(out))
+    abspath = os.path.abspath(out)
+    dpath = os.path.dirname(abspath)
     tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
     try:
         with external_repo(cache_dir=tmp_dir, url=url, rev=rev) as repo:
-            # Note: we need to replace state, because in case of getting DVC
-            # dependency on CIFS or NFS filesystems, sqlite-based state
-            # will be unable to obtain lock
-            repo.state = StateNoop()
+            tmp_repo_path = os.path.join(repo.root_dir, path)
+            if os.path.exists(tmp_repo_path):
+                if os.path.isdir(tmp_repo_path):
+                    shutil.copytree(tmp_repo_path, abspath)
+                else:
+                    shutil.copy2(tmp_repo_path, abspath)
```

Comment:
Why can't we simply move it?

Reply:
Good idea

Reply:
@Suor But our git repos are cached, moving those files will corrupt the cache. Or am I missing something?

Reply:
Hmm, yes it will. Let's copy then. it shouldn't be too big if it stored in git anyway.